bundles/org.openhab.binding.teslapowerwall/src/main/java/org/openhab/binding/teslapowerwall/internal/TeslaPowerwallHandler.java

```diff
@@ -0,0 +1,183 @@
+import org.openhab.binding.teslapowerwall.internal.api.BatterySOE;
+import org.openhab.binding.teslapowerwall.internal.api.GridStatus;
+import org.openhab.binding.teslapowerwall.internal.api.MeterAggregates;
+import org.openhab.binding.teslapowerwall.internal.api.Operations;
+import org.openhab.binding.teslapowerwall.internal.api.SystemStatus;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.MetricPrefix;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TeslaPowerwallHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Paul Smedley - Initial contribution
+ */
+@NonNullByDefault
+public class TeslaPowerwallHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(TeslaPowerwallHandler.class);
+
+    private @NonNullByDefault({}) TeslaPowerwallConfiguration config;
+    private @NonNullByDefault({}) TeslaPowerwallWebTargets webTargets;
+    private @Nullable ScheduledFuture<?> pollFuture;
+
+    public TeslaPowerwallHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        config = getConfigAs(TeslaPowerwallConfiguration.class);
+        webTargets = new TeslaPowerwallWebTargets(config.hostname, httpClient);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.warn("This binding is read only");
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(TeslaPowerwallConfiguration.class);
+        logger.debug("config.hostname = {}, refresh = {}", config.hostname, config.refresh);
+        if (config.hostname.isBlank() || config.email.isBlank() || config.password.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error.missing-config-key");
+            return;
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+            schedulePoll();
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private void schedulePoll() {
+        ScheduledFuture<?> pollFuture = this.pollFuture;
+        if (pollFuture != null) {
+            pollFuture.cancel(false);
+        }
+        logger.debug("Scheduling poll for 1 second out, then every {} s", config.refresh);
+        this.pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 1, config.refresh, TimeUnit.SECONDS);
```

Comment:
```suggestion
        this.pollFuture = scheduler.scheduleWithFixedDelay(this::pollStatus, 1, config.refresh, TimeUnit.SECONDS);
```
`poll` is a no longer needed wrapper.

Reply:
Fixed.